Trappist/src/Promact.Trappist.Repository/Questions/QuestionRepository.cs

```diff
@@ -44,29 +47,47 @@ public ICollection<SingleMultipleAnswerQuestionApplicationClass> GetAllQuestions
             _dbContext.SaveChanges();
             return(questionAC);
         }
+
         /// <summary>
         /// Add new code snippet question to the database
         /// </summary>
         /// <param name="codeSnippetQuestion">Code Snippet Question Model</param>
-        public void AddCodeSnippetQuestion(CodeSnippetQuestionDto codeSnippetQuestionModel)
+        public async Task AddCodeSnippetQuestionAsync(QuestionAC questionAC)
         {
-            CodeSnippetQuestion codeSnippetQuestion = Mapper.Map<CodeSnippetQuestionDto, CodeSnippetQuestion>(codeSnippetQuestionModel);
+            var codeSnippetQuestionModel = questionAC.CodeSnippetQuestionAC;
+            CodeSnippetQuestion codeSnippetQuestion = Mapper.Map<CodeSnippetQuestionAC, CodeSnippetQuestion>(codeSnippetQuestionModel);
+
             using (var transaction = _dbContext.Database.BeginTransaction())
             {
-                var question = _dbContext.CodeSnippetQuestion.Add(codeSnippetQuestion);
-                _dbContext.SaveChanges();
-                var codingLanguageList = codeSnippetQuestionModel.LanguageList;
-                //To-Do Change according to new model var questionId = question.Entity.Id; 
+                //Add common question details
+                var question = await _dbContext.Question.AddAsync(questionAC.Question);
+                await _dbContext.SaveChangesAsync();
+
+                var questionId = question.Entity.Id;
+
+                //Add codeSnippet part of question
+                codeSnippetQuestion.QuestionId = questionId;
+                var codingQuestion = await _dbContext.CodeSnippetQuestion.AddAsync(codeSnippetQuestion);
```

Comment:
codingQuestion is used to get Id generated after saveChanges in CodeSnippetQuestion table.